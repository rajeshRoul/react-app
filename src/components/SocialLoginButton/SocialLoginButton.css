.SocialButton{
    background-color: transparent;
    outline: none;
    border: none;
}

.SocialButton img{
    width: 50px;
    height: 50px;
}