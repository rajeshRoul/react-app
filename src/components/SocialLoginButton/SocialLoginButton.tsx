import './SocialLoginButton.css'
import React, { ReactElement } from 'react'

type SocialButtonPropType = {
    handleClick: () => void
    logoUrl: string
    name: string
}

function SocialLoginButton(props: SocialButtonPropType): ReactElement {
    const { handleClick, logoUrl, name } = props
    return (
        <button
            data-testid="socialButton"
            className="SocialButton"
            type="button"
            onClick={handleClick}
        >
            <img src={logoUrl} alt={name} />
        </button>
    )
}

export default SocialLoginButton
