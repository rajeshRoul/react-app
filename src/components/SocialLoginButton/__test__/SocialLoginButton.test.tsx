/* eslint-disable no-undef */
import React from 'react'
import { render, cleanup, fireEvent, screen } from '@testing-library/react'
import SocialLoginButton from '../SocialLoginButton'
import '@testing-library/jest-dom/extend-expect'

afterEach(cleanup)

test('social login button click', () => {
    const handleClick = jest.fn()
    const { getByTestId } = render(
        <SocialLoginButton
            name="test"
            logoUrl="/test"
            handleClick={handleClick}
        />
    )
    fireEvent.click(getByTestId('socialButton'))
    expect(handleClick).toHaveBeenCalledTimes(1)
    fireEvent.click(getByTestId('socialButton'))
    expect(handleClick).toHaveBeenCalledTimes(2)
    const imgElement = screen.getByAltText('test')
    expect(imgElement).toBeInTheDocument()
})

it('should take a snapshot', () => {
    const { getByTestId } = render(
        <SocialLoginButton name="test" logoUrl="/test" handleClick={() => {}} />
    )
    expect(getByTestId('socialButton')).toMatchSnapshot()
})
