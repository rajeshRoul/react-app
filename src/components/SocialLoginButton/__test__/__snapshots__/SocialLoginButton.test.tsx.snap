// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<button
  class="SocialButton"
  data-testid="socialButton"
  type="button"
>
  <img
    alt="test"
    src="/test"
  />
</button>
`;
